src/utils/address-utils.ts

```diff
@@ -1092,34 +1092,37 @@ function getAddress(
   section: string,
   addressCase: AddressCases,
   addressHierarchy: string[],
-  conditionalCase?: string
+  conditionalCase?: string | Conditional[]
 ): SerializedFormField[] {
   const defaultFields: SerializedFormField[] = getAddressFields(
     section,
     addressCase,
     addressHierarchy
   )
   if (conditionalCase) {
-    defaultFields.forEach((field) => {
-      let conditional
-      if (conditionalCase) {
-        conditional = {
-          action: 'hide',
-          expression: `${conditionalCase}`
-        }
-      }
-      if (
-        conditional &&
-        field.conditionals &&
-        field.conditionals.filter(
-          (conditional) => conditional.expression === conditionalCase
-        ).length === 0
-      ) {
-        field.conditionals.push(conditional)
-      } else if (conditional && !field.conditionals) {
-        field.conditionals = [conditional]
-      }
-    })
+    return addConditionalsToFields(defaultFields, conditionalCase)
   }
+
   return defaultFields
 }
+
+/**
+ * Adds provided conditionals to each field. Mutates the given array.
+ */
+function addConditionalsToFields(
+  fields: SerializedFormField[],
+  conditionalCase: string | Conditional[]
+) {
+  fields.forEach((field) => {
```

Comment:
Was there some other mutation happening on a higher level that forced this function also to mutate the inner array values?

Reply:
I will check! I opted for caution if there was some performance reason for mutating.

Reply:
If you see code with side effects in our codebase you can safely assume it's either:
1. Something higher in the call stack is built using mutation and the author didn't have time to fix the root issue
2. Side-effectful code got past code review unnoticed

We advocate for immutable values and anything else should be a rare exception

Reply:
Clear! On this particular case, We are mutating on higher levels. Taking into account the 1.5. delivery, I suggest I'll merge this one after the new changes have been reviewed and check whether I could get rid of the mutations.